core/src/main/scala/com/salesforce/op/stages/impl/feature/TextTokenizer.scala

```diff
@@ -151,8 +151,8 @@ object TextTokenizer {
    * @param minTokenLength      minimum token length
    * @return detected language and sentence tokens
    */
-  def tokenize(
-    text: Text,
+  def tokenizeString(
+    textString: String,
```

Comment:
now this function can explode with NullPointerException is textString is null, while before it could have happened.

Reply:
Hmmm, I'm not sure if it's possible for the `textString` argument to be `null` in practice though. When this function used for tokenizing the map entries, a value that was originally null there will just not show up as an entry in the map. When it's used for tokenizing a normal Text entry, then we should have already safely converted any nulls or missing elements into an `Option[String]`, right?

The actual tokenize call during vectorization is still `tokenize(v.toText)` where `v` is the value in a text map. I'd actually argue that that should be changed to `tokenizeText(v)` to save time converting it to Text and back again.

I agree it's technically less safe, but I don't think it's necessary to have null checking at this point in the flow. We should make sure the data gets created in a safe way, which I think we already do. Are there some specific edge cases I'm missing?

Reply:
I think the simplest is to add a null check `tokenizeString` and return

Reply:
Now that I think about it more, I'm pretty sure the old tokenize function would also give a NPE if you fed it a sneaky null value. The `SomeValue.unapply` function explicitly calls `v.isEmpty` which would also fail if v was `null`.

I put back the old tokenize function as `oldTokenize` and tried
```val sneakyStringOpt: Option[String] = null
val myText = Text(sneakyStringOpt)
val res = TextTokenizer.oldTokenize(myText)```
which did indeed throw a NPE.

We have tests all over the place (eg. our vectorizer tests and FeatureTypeSparkConverterTest) that make sure we can handle `null` values in dataframes and safely convert them into our types. I'm not aware of any explicit null checks in our functions elsewhere, so it just feels weird to put one here.

@leahmcguire any opinions on this?

Reply:
SomeValue.unapply operates on value which is Option[String]. Null check is done during the construction of Text.

Reply:
Your example you provided is not currently possible and also not a fair one :)

Reply:
@Jauntbox is this only called from the Option[String] version below? if so make it private and it is fine.

Reply:
In fact make them both private please